### ISSUE TITLE
Unable to connect to BigTable emulator with v2.0.0

### ISSUE THREAD

**Issue by nielsbasjes** [googleapis/java-bigtable-hbase#3518]
TL;DR:
A working application using bigtable-hbase-2.x 1.26.3 stops working after updating to 2.0.0 with 
```java.lang.IllegalStateException: BigtableInstanceAdminSettings doesn't supported on Emulator```

#### Environment details

1. Specify the API at the beginning of the title. General, Core, and Other are also allowed as types
2. OS type and version: Ubuntu 20.04 LTS
3. Java version: 11.0.13
4. Version(s): 1.26.3 (works) and 2.0.0 (does not work) 

#### Steps to reproduce

Make a junit test in Java with the code fragment provided below.
Use the dependency com.google.cloud.bigtable:bigtable-hbase-2.x:1.26.3

Output:
```
We have a connection: AbstractBigtableConnection{project=MyTestProject, instance=HBaseTestingInstance, dataHost=127.0.0.1, tableAdminHost=127.0.0.1}
```

Now change the dependency to  com.google.cloud.bigtable:bigtable-hbase-2.x:2.0.0
and you get the exception (stacktrace below)
```java.lang.IllegalStateException: BigtableInstanceAdminSettings doesn't supported on Emulator```

#### Code example

```java
        Configuration config = BigtableConfiguration.configure("MyTestProject", "HBaseTestingInstance");
        config.set(BIGTABLE_EMULATOR_HOST_KEY, getDockerIpAddress() + ":" + getDockerBigtablePort());

        try (Connection connection = ConnectionFactory.createConnection(config)) {
            LOG.info("We have a connection: {}", connection);
        }
```

#### Stack trace
```
Caused by: java.lang.IllegalStateException: BigtableInstanceAdminSettings doesn't supported on Emulator
	at com.google.common.base.Preconditions.checkState(Preconditions.java:502)
	at com.google.cloud.bigtable.admin.v2.BigtableInstanceAdminSettings.newBuilder(BigtableInstanceAdminSettings.java:118)
	at com.google.cloud.bigtable.hbase.wrappers.veneer.BigtableHBaseVeneerSettings.buildBigtableInstanceAdminSettings(BigtableHBaseVeneerSettings.java:369)
	at com.google.cloud.bigtable.hbase.wrappers.veneer.BigtableHBaseVeneerSettings.<init>(BigtableHBaseVeneerSettings.java:168)
	at com.google.cloud.bigtable.hbase.wrappers.veneer.BigtableHBaseVeneerSettings.create(BigtableHBaseVeneerSettings.java:156)
	at com.google.cloud.bigtable.hbase.wrappers.BigtableHBaseSettings.create(BigtableHBaseSettings.java:42)
	at org.apache.hadoop.hbase.client.AbstractBigtableConnection.<init>(AbstractBigtableConnection.java:111)
	at org.apache.hadoop.hbase.client.AbstractBigtableConnection.<init>(AbstractBigtableConnection.java:86)
	at com.google.cloud.bigtable.hbase2_x.BigtableConnection.<init>(BigtableConnection.java:61)
	... 41 more
```

#### External references such as API reference guides

- ?
 
#### Any additional information below

Following these steps guarantees the quickest resolution possible.

Thanks!

**Comment by nielsbasjes** [comment]
Additional info:
I found there are two ways to indicate the BigTable emulator is to be used when looking at the various clients.
- Set the environment variable BIGTABLE_EMULATOR_HOST
- In the HBase config set the key BIGTABLE_EMULATOR_HOST_KEY

I have a library written as a wrapper around the provided emulator docker image to make it a lot easier to use so for most clients. So my code must set the BIGTABLE_EMULATOR_HOST for testing, for the HBase client I need to set the BIGTABLE_EMULATOR_HOST_KEY.

In the code of this client it turns out that it is quite inconsistent in this area:
https://github.com/googleapis/java-bigtable-hbase/blob/main/bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java#L366
```
    String emulatorEndpoint = configuration.get(BIGTABLE_EMULATOR_HOST_KEY);
    if (!Strings.isNullOrEmpty(emulatorEndpoint)) {
      // todo switch to newBuilderForEmulator once available
      adminBuilder = BigtableInstanceAdminSettings.newBuilder();
```

and the call to create the adminBuilder does this:
```
    public static BigtableInstanceAdminSettings.Builder newBuilder() {
        Preconditions.checkState(System.getenv("BIGTABLE_EMULATOR_HOST") == null, "BigtableInstanceAdminSettings doesn't supported on Emulator");
        return new BigtableInstanceAdminSettings.Builder();
    }
```

So apparently the properties and checks across all of this are inconsistent.
Also because if I unset the BIGTABLE_EMULATOR_HOST the HBase client works again ... only to make the other clients fail.

**Comment by nielsbasjes** [comment]
At this point my current assessment is that implementing the mentioned "// todo switch to newBuilderForEmulator once available" seems like the correct solution.
